circletree/CircleSegment.jsx

```diff
@@ -139,7 +140,7 @@ var CircleSegment = React.createClass({
       updateBBox: this.updateBBox
     };
 
-    return this.props.data.map( (type, pos) => {
+    return this.props.data.filter( (type) => !this.props.filter[type] ).map( (type, pos) => {
```

Comment:
no need for `(type)` if there's only one argument. `.filter(type => !this.props.filter[type])` will work